examples/demo.ipynb

```diff
@@ -486,13 +486,18 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "Guppy compilation failed. Error in file <In [14]>:6\n",
+      "Error: Illegal assignment (at <In [14]>:6:8)\n",
+      "  | \n",
+      "4 | def outer(x: int) -> int:\n",
+      "5 |     def nested() -> None:\n",
+      "6 |         x += 1  # Mutation of captured variable x is not allowed\n",
+      "  |         ^^^^^^ Variable `x` may not be assigned to since `x` is captured\n",
+      "  |                from an outer scope\n",
+      "  | \n",
+      "4 | def outer(x: int) -> int:\n",
+      "  |           ------ `x` defined here\n",
```

Comment:
It would be cool to detect overlapping spans and only print the `outer` once like
```
def outer(x: int) -> int
                ----- `x` defined here
    def nested() -> None
        x += 1
        ^^^^^^ Variable `x`...
```
but that seems tricky...

Reply:
Yes, good idea, Rust also does this! Added to #608